File: frontend/src/api/dto/configDto.ts
import type { MapCenterDto } from './mapCenterDto.ts'
import type { MapStyleDto } from './mapStyleDto.ts'
import type { MapTerrainDto } from './mapTerrainDto.ts'

/**
 * Application configuration
 */
export interface ConfigDto {
  /** WebAuthn Relying Party ID (effective host) */
  webAuthnRpId: string
  /** Application name */
  appName: string
  /** Single team mode - team creation disabled */
  singleTeam: boolean
  /** Whether the interactive planner is open in the team-independent GPX tools. When false the tools still accept a .gpx upload, only drawing from scratch is closed. Platform-admin only, per domain. */
  enableGpxPlanner: boolean
  /** Slug of the team the site is pinned to (dedicated hostname / alias). Null on a regular multi-team domain. When set, the app roots on this team. */
  pinnedTeamSlug?: string
  /** Basemaps the clients may offer, in switcher order. Served rather than compiled in, so a tile provider can change without a client release. */
  mapStyles: MapStyleDto[]
  /** Public base URL of the tile host, for a client that builds its own style, sprite or glyph URLs. */
  tileServerBaseUrl: string
  /** Where a map opens before it knows what it is showing. On a site rooted on one team this is that team's location; otherwise the deployment default. */
  defaultCenter: MapCenterDto
  /** The elevation source the clients may shade the relief with. Null when the deployment configures none — the clients then offer no relief at all rather than falling back to a provider of their own. */
  terrain?: MapTerrainDto
  /** Oldest mobile build this server still serves, as a semver string. Null when no floor is enforced; a client older than this should tell the user to update. */
  minSupportedAppVersion?: string
}

File: backend/src/main/java/fr/pedalons/dto/config/ConfigDto.java
package fr.pedalons.dto.config;

import fr.pedalons.dto.validation.ValidateSchema;
import java.util.List;
import org.eclipse.microprofile.openapi.annotations.media.Schema;
import org.jspecify.annotations.Nullable;

@Schema(description = "Application configuration")
@ValidateSchema
public record ConfigDto(
    @Schema(description = "WebAuthn Relying Party ID (effective host)", required = true)
        String webAuthnRpId,
    @Schema(description = "Application name", required = true) String appName,
    @Schema(description = "Single team mode - team creation disabled", required = true)
        boolean singleTeam,
    @Schema(
            description =
                "Whether the interactive planner is open in the team-independent GPX tools. When"
                    + " false the tools still accept a .gpx upload, only drawing from scratch is"
                    + " closed. Platform-admin only, per domain.",
            required = true)
        boolean enableGpxPlanner,
    @Schema(
            description =
                "Slug of the team the site is pinned to (dedicated hostname / alias). Null on a"
                    + " regular multi-team domain. When set, the app roots on this team.")
        @Nullable String pinnedTeamSlug,
    @Schema(
            description =
                "Basemaps the clients may offer, in switcher order. Served rather than compiled in,"
                    + " so a tile provider can change without a client release.",
            required = true)
        List<MapStyleDto> mapStyles,
    @Schema(
            description =
                "Public base URL of the tile host, for a client that builds its own style, sprite"
                    + " or glyph URLs.",
            required = true)
        String tileServerBaseUrl,
    @Schema(
            description =
                "Where a map opens before it knows what it is showing. On a site rooted on one team"
                    + " this is that team's location; otherwise the deployment default.",
            required = true)
        MapCenterDto defaultCenter,
    @Nullable
        @Schema(
            description =
                "The elevation source the clients may shade the relief with. Null when the"
                    + " deployment configures none — the clients then offer no relief at all rather"
                    + " than falling back to a provider of their own.")
        MapTerrainDto terrain,
    @Nullable
        @Schema(
            description =
                "Oldest mobile build this server still serves, as a semver string. Null when no"
                    + " floor is enforced; a client older than this should tell the user to"
                    + " update.")
        String minSupportedAppVersion) {}
